mailpoet/lib/Subscribers/ImportExport/PersonalDataExporters/SubscriberExporter.php

```diff
@@ -2,94 +2,132 @@
+      'data' => $this->exportSubscriber($this->subscribersRepository->findOneBy(['email' => trim($email)])),
       'done' => true,
     ];
   }
 
-  private function exportSubscriber($subscriber) {
+  /**
+   * @param SubscriberEntity|null $subscriber
+   * @return array|mixed[][]
+   */
+  private function exportSubscriber(?SubscriberEntity $subscriber): array {
     if (!$subscriber) return [];
     return [[
       'group_id' => 'mailpoet-subscriber',
       'group_label' => WPFunctions::get()->__('MailPoet Subscriber Data', 'mailpoet'),
-      'item_id' => 'subscriber-' . $subscriber->id,
-      'data' => $this->getSubscriberExportData($subscriber->withCustomFields()),
+      'item_id' => 'subscriber-' . $subscriber->getId(),
+      'data' => $this->getSubscriberExportData($subscriber),
     ]];
   }
 
-  private function getSubscriberExportData($subscriber) {
+  /**
+   * @param SubscriberEntity $subscriber
+   * @return mixed[][]
+   */
+  private function getSubscriberExportData(SubscriberEntity $subscriber): array {
     $customFields = $this->getCustomFields();
     $result = [
       [
         'name' => WPFunctions::get()->__('First Name', 'mailpoet'),
-        'value' => $subscriber->firstName,
+        'value' => $subscriber->getFirstName(),
       ],
       [
         'name' => WPFunctions::get()->__('Last Name', 'mailpoet'),
-        'value' => $subscriber->lastName,
+        'value' => $subscriber->getLastName(),
       ],
       [
         'name' => WPFunctions::get()->__('Email', 'mailpoet'),
-        'value' => $subscriber->email,
+        'value' => $subscriber->getEmail(),
       ],
       [
         'name' => WPFunctions::get()->__('Status', 'mailpoet'),
-        'value' => $subscriber->status,
+        'value' => $subscriber->getStatus(),
       ],
     ];
-    if ($subscriber->subscribedIp) {
+    if ($subscriber->getSubscribedIp()) {
       $result[] = [
         'name' => WPFunctions::get()->__('Subscribed IP', 'mailpoet'),
-        'value' => $subscriber->subscribedIp,
+        'value' => $subscriber->getSubscribedIp(),
       ];
     }
-    if ($subscriber->confirmedIp) {
+    if ($subscriber->getConfirmedIp()) {
       $result[] = [
         'name' => WPFunctions::get()->__('Confirmed IP', 'mailpoet'),
-        'value' => $subscriber->confirmedIp,
+        'value' => $subscriber->getConfirmedIp(),
       ];
     }
     $result[] = [
       'name' => WPFunctions::get()->__('Created at', 'mailpoet'),
-      'value' => $subscriber->createdAt,
+      'value' => $subscriber->getCreatedAt()
+        ? $subscriber->getCreatedAt()->format(DateTime::DEFAULT_DATE_TIME_FORMAT)
+        : '',
     ];
 
-    foreach ($customFields as $customFieldId => $customFieldName) {
-      $customFieldValue = $subscriber->{$customFieldId};
-      if ($customFieldValue) {
+    foreach ($subscriber->getSubscriberCustomFields() as $field) {
+      if (isset($this->getCustomFields()[$field->getId()])) {
         $result[] = [
```

Comment:
This also changes the logic. The old code was skipping custom fields with falsy values (`if ($customFieldValue)`). The new code checks if the custom field is defined in the list of all custom fields. Is this change intentional?

Thinking about it. The change might be ok because the current export is skipping falsy values, but they also carry information.

Reply:
Maybe we could consult this ⬆️  with @pavel-mailpoet .

Reply:
I think it is fine to keep those custom fields with false values in the export.